docs/src/further_topics/ugrid/data_model.rst

```diff
@@ -0,0 +1,328 @@
+retain their familiar structured format.
+
+The UGRID conventions themselves are designed as an addition to the existing CF
+conventions, which are at the core of Iris' philosophy.
+
+What's Different?
+=================
+
+The mesh format represents data's geographic location using an **unstructured
+mesh**. This has significant pros and cons when compared to a structured grid.
+
+.. contents::
+   :local:
+
+The Detail
+----------
+..
+    The diagram images are SVG's, so editable by any graphical software
+     (e.g. Inkscape). They were originally made in MS PowerPoint.
+
+    Uses the IBM Colour Blind Palette (see
+     http://ibm-design-language.eu-de.mybluemix.net/design/language/resources/color-library
+     )
+
+Structured Grids
+~~~~~~~~~~~~~~~~
+Assigning data to locations using a structured grid is essentially an act of
+matching coordinate arrays to each dimension of the data array. The data can
+also be represented as an area (instead of a point) by including a bounds array
+for each coordinate array.
+
+.. figure:: images/data_structured_grid.svg
+   :alt: Diagram of how data is represented on a structured grid
+
+   Data on a structured grid
+
+   :download:`full size image <images/data_structured_grid.svg>`
+
+Unstructured Meshes
+~~~~~~~~~~~~~~~~~~~
+The most basic element in a mesh is the 0-dimensional **node**: a single
+location in space. Every node in the mesh is defined by indexing the
+1-dimensional X and Y (and optionally Z) coordinate arrays (the
+``node_coordinates``) - e.g. ``(x[3], y[3])`` gives the position of the fourth
+node. Note that this means each node has its own coordinates, independent of
+every other node. Since nodes can be anywhere in the space - **unstructured** -
+the position in the array has nothing to do with spatial position.
+
+If data is assigned to node location it must be stored in a 1D array of equal
+length to the number of nodes in the mesh. ``data[3]`` is at the position:
+``(x[3], y[3])``.
+
+Data can also be assigned to higher dimensional elements - **edges**, **faces**
+or **volumes**. These elements are constructed by connecting nodes together
+using a 2-dimensional **connectivity** array. One dimension varies over each
+element, while the other dimension varies over each node that makes up that
+element; the values in the array are the node indices. E.g. we could make 2
+square faces from 6 nodes using this ``face_node_connectivity``:
+``[[0, 1, 3, 2], [2, 3, 5, 4]]``. Remember that meshes are **unstructured**, so
+there is no significance to the order of the faces in the array. Data assigned
+to a higher dimensional location must be stored in a 1D array of equal length
+to the number of elements in the mesh, e.g. ``my_data = [0.33, 4.02]`` for our
+example.
+
+.. note::
+
+        Connectivities also exist to connect the higher dimensional elements,
+        e.g. ``face_edge_connectivity``. These are optional conveniences to
```

Comment:
Do we not hit problems where only having node-face connectivity without node-edge and edge-face means that we define faces ambiguously because we could reorder the nodes used? Or does order in the connectivity array define that for us? Also, how do we define "inside"?

Reply:
>Or even that the face coords always lie strictly within the face ??

Yes I had that same moment of pause while writing the docs 😅

Reply:
I can address enough of the uncertainty here by mentioning the anti-clockwise thing.